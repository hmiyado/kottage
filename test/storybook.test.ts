import initStoryshots, {
  Stories2SnapsConverter,
} from '@storybook/addon-storyshots'
import { act } from '@testing-library/react'
import { setupServer } from 'msw/node'
import React from 'react'
import { create } from 'react-test-renderer'

//https://github.com/storybookjs/storybook/issues/8083#issuecomment-533184737
jest.mock('global', () =>
  Object.assign(global, { window: { STORYBOOK_HOOKS_CONTEXT: '' } })
)

// https://github.com/storybookjs/storybook/issues/7745#issuecomment-807338022
const converter = new Stories2SnapsConverter()
initStoryshots({
  test: async ({ story, context, done }) => {
    const endpoints: any[] = context.parameters.msw?.handlers ?? []
    const shouldMockEndpoints = endpoints.length > 0
    const server = setupServer(...endpoints)
    server.listen()

    try {
      let renderer
      act(() => {
        // React.createElement() is important because of hooks [shouldn't call story.render() directly]
        renderer = create(React.createElement(story.render))
      })

      // wait for state changes
      await act(
        () =>
          new Promise((resolve) =>
            setTimeout(resolve, shouldMockEndpoints ? 10 : 0)
          )
      )

      expect(renderer).toMatchSnapshot()

      done && done()
    } finally {
      server.close()
    }
  },
})
