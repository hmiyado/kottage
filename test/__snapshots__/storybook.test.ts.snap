// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Contact Primary 1`] = `
<section
  className="section"
>
  <h2
    className="headline"
  >
    Skill
  </h2>
  <p
    className="body"
  >
    Android/Kotlin エンジニア。その他 Swift/iOS JavaScript, TypeScript, Ruby など。詳細は
     
    <a
      className="link"
      href="https://github.com/hmiyado/resume"
    >
      GitHub レジュメ
    </a>
     
    を参照。
  </p>
  <h2
    className="headline"
  >
    Price
  </h2>
  <p
    className="body"
  >
    時間単価6000円〜。詳細要相談。
  </p>
  <h2
    className="headline"
  >
    Contact
  </h2>
  <p
    className="body"
  >
    <a
      className="link"
      href="https://twitter.com/miyado20th"
    >
      Twitter
    </a>
  </p>
</section>
`;

exports[`Storyshots Entry Entry 1 1`] = `
Array [
  <h2
    className="headline"
  >
    GW
  </h2>,
  <p
    className="body"
  >
    GW はコーディングしたり、リモートゲームしたり、無計画だったわりには楽しく過ごせた。コーディングでは2年前のリポジトリを引っ張り出して整理したら、思っていた以上に興が乗った。次に触るのは2年後のGWかもしれない。
  </p>,
  <div
    className="footer"
  >
    <div
      className="text"
    >
      2021-05-05T21:12
    </div>
    <div
      className="text"
    >
      hmiyado
    </div>
  </div>,
]
`;

exports[`Storyshots Entry Primary 1`] = `
Array [
  <h2
    className="headline"
  >
    short title
  </h2>,
  <p
    className="body"
  >
    body
  </p>,
  <div
    className="footer"
  >
    <div
      className="text"
    >
      2021-05-05T21:12
    </div>
    <div
      className="text"
    >
      hmiyado
    </div>
  </div>,
]
`;

exports[`Storyshots Sentence Long 1`] = `
Array [
  <h2
    className="headline"
  >
    long long long long long long title
  </h2>,
  <p
    className="body"
  />,
]
`;

exports[`Storyshots Sentence Short 1`] = `
Array [
  <h2
    className="headline"
  >
    short title
  </h2>,
  <p
    className="body"
  />,
]
`;
